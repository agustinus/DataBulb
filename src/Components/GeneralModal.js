import React, {Component} from 'react';
import {Modal, Text, TouchableOpacity, View, StyleSheet} from 'react-native';
import PropTypes from 'prop-types';

export default class GeneralModal extends Component {
  constructor(props) {
    super(props);
    this.state = {
      modalVisible: this.props.isVisible,
    };
  }

  static propTypes = {
    title: PropTypes.string,
    content: PropTypes.string,
    onDismiss: PropTypes.func,
  };

  render() {
    return (
      <Modal
        animationType="fade"
        transparent={true}
        visible={this.state.modalVisible}
        presentationStyle="overFullScreen"
        onDismiss={this.props.onDismiss}>
        <View style={styles.background}>
          <View style={styles.alertDialog}>
            <View style={styles.contentWrapper}>
              <Text style={styles.title}>{this.props.title}</Text>
              <Text style={styles.content}>{this.props.content}</Text>
            </View>
            <TouchableOpacity
              style={styles.buttonWrapper}
              onPress={() => this._setModalVisible(!this.state.modalVisible)}>
              <Text style={styles.buttonText}>OK</Text>
            </TouchableOpacity>
          </View>
        </View>
      </Modal>
    );
  }

  _setModalVisible(visible) {
    this.setState({modalVisible: visible});
  }
}

const styles = StyleSheet.create({
  title: {fontSize: 14, color: '#444444', textAlign: 'center'},
  content: {
    fontSize: 36,
    color: '#000000',
    textAlign: 'center',
    fontWeight: 'bold',
  },
  buttonText: {
    fontSize: 20,
    color: '#868686',
    textAlign: 'center',
    fontWeight: 'bold',
  },
  buttonWrapper: {flex: 1, justifyContent: 'center'},
  contentWrapper: {
    flex: 2,
    justifyContent: 'space-around',
    paddingTop: 30,
    paddingBottom: 5,
  },
  alertDialog: {
    height: 200,
    width: 300,
    backgroundColor: 'white',
    borderRadius: 8,
    flexDirection: 'column',
    alignItems: 'stretch',
  },
  background: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#000000AA',
  },
});
