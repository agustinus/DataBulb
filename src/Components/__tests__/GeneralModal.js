import React from 'react';
import GeneralModal from '../GeneralModal';
import Adapter from 'enzyme-adapter-react-16';
import {configure, shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import ReactTestRenderer from 'react-test-renderer';

configure({adapter: new Adapter()});

describe('GeneralModal', () => {
  let wrapper = null;
  let instance = null;

  beforeAll(() => {
    const props = {isVisible: false, onDismiss: jest.fn()};
    wrapper = shallow(<GeneralModal {...props} />);
    instance = wrapper.instance();
  });

  it('renders correctly', () => {
    expect(
      ReactTestRenderer.create(<GeneralModal />).toJSON(),
    ).toMatchSnapshot();
  });

  it('should invoke _setModalVisible() when onPress', () => {
    wrapper
      .find('TouchableOpacity')
      .props()
      .onPress(true);
    expect(wrapper.state().modalVisible).toEqual(true);
    wrapper
      .find('TouchableOpacity')
      .props()
      .onPress(true);
    expect(instance.props.onDismiss).toHaveBeenCalledTimes(1);
  });

  it('should show button text OK if no buttonText props set', () => {
    const buttonText = wrapper.findWhere(
      node => node.prop('testID') === 'testIDButtonText',
    );
    expect(buttonText.exists()).toBe(true);
    expect(toJson(buttonText)).toMatchSnapshot();
  });

  it('should show button text from set props', () => {
    wrapper.setProps({buttonText: 'This new button text'});
    const buttonText = wrapper.findWhere(
      node => node.prop('testID') === 'testIDButtonText',
    );
    expect(buttonText.exists()).toBe(true);
    expect(toJson(buttonText)).toMatchSnapshot();
  });
});
