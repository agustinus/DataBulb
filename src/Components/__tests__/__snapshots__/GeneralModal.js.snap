// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeneralModal renders correctly 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  presentationStyle="overFullScreen"
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#000000AA",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "stretch",
          "backgroundColor": "white",
          "borderRadius": 8,
          "flexDirection": "column",
          "height": 200,
          "width": 300,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 2,
            "justifyContent": "space-around",
            "paddingBottom": 5,
            "paddingTop": 30,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#444444",
              "fontSize": 14,
              "textAlign": "center",
            }
          }
        />
        <Text
          style={
            Object {
              "color": "#000000",
              "fontSize": 36,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        />
      </View>
      <View
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#868686",
              "fontSize": 20,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
          testID="testIDButtonText"
        >
          OK
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`GeneralModal should show button text OK if no buttonText props set 1`] = `
<Text
  style={
    Object {
      "color": "#868686",
      "fontSize": 20,
      "fontWeight": "bold",
      "textAlign": "center",
    }
  }
  testID="testIDButtonText"
>
  OK
</Text>
`;

exports[`GeneralModal should show button text from set props 1`] = `
<Text
  style={
    Object {
      "color": "#868686",
      "fontSize": 20,
      "fontWeight": "bold",
      "textAlign": "center",
    }
  }
  testID="testIDButtonText"
>
  This new button text
</Text>
`;
